### DISCUSSION TITLE
Questions about "Jukebox" mode

### DISCUSSION THREAD

**Discussion by GK-774** [jcorporation/myMPD#603] [Q&A] [answered]
myMPD version: 8.1.5 (devel from 6.11.)

So wanted to try Jukebox which I have not used before, clicked "Jukebox" to change the view from "Queue" and got "Jukebox is disabled".
Traversed through the settings but "Jukebox" is nowhere mentioned. The documentation is rather quiet about it, so this are my questions:

What is it supposed to do? (Compared to a "hardware" jukebox adding a song to the queue already does the same, without insering a coin of course.)

Where would one enable it, to give it a try?

If it is disabled, would it not make sense to hide the "Jukebox" button as well?

Sorry for the inconvenience, but I really could not figure out how this supposed to work.

**Comment by GK-774** [comment]
Oh, sorry - my fault. It´s in the playback settings.

**Comment by jcorporation** [comment]
Please ask questions under discussion not in issues.

**Comment by GK-774** [comment]
Well now it looks like a bug. Enabled it in "playback settings" but still get "Jukebox is disabled" no matter if I try Album or Song (Database or Playlist). It gets not enabled.

**Comment by GK-774** [comment]
Sorry to have posted it under issues. So to rephrase my questions from above:

What is it supposed to do? (Compared to a "hardware" jukebox adding a song to the queue already does the same, without insering a coin of course.)

Where would one enable it, to give it a try? (enabling it Playback Settings does not change the "Jukebox is disabled" message.

If it is disabled, would it not make sense to hide the "Jukebox" button as well?

**Comment by jcorporation** [comment, marked as answer]
- Enabling the jukebox mode ads randoms songs to the queue some seconds before the queue length gets smaller than the "Keep queue length" setting.
- Consume must be enabled for this to work.
- Jukebox view is automatically populated after the jukebox triggers the first time.

**Reply by GK-774** [reply]
Ok, understood. Thanks for the explanation and yes works as described. The "disabled" instead of "not triggered yet" message is a bit confusing, although correct in a technical sense, when it´s not triggered it´s still "disabled".

**Comment by MiM-TMassey** [comment]
For those who are struggling with how to turn on Jukebox mode (like me):  you need to enable it from the Queue page, not from the main MyMPD settings page.

Use the first icon to the right of the search bar (playback settings):
![image](https://user-images.githubusercontent.com/35174197/180917957-8746ef13-8064-4742-b66d-cb32f253e2c1.png)

Then use the Jukebox tab to enable and configure it:
![image](https://user-images.githubusercontent.com/35174197/180918083-6d409534-8f08-483b-be97-2a8919df864c.png)

Hope that saves someone else some frustration!

**Comment by sipaktli** [comment]
Well, I could not figure out on the interface as of the screenshot above or any menus, so I enabled it manually:

echo "1" > ~/.config/mympd/state/default/jukebox_mode
echo "7" > ~/.config/mympd/state/default/jukebox_queue_length

**Comment by john-** [comment]
It is in the bottom right on every screen:
<img width="335" height="113" alt="image" src="https://github.com/user-attachments/assets/a4fb830c-f687-43d7-863d-f74970c2f134" />

From there select the Jukebox tab.

The up arrow next to it is for presents.

**Comment by chiefy** [comment]
Not sure what happened, but I have Jukebox mode enabled / consume / set to Album mode. Was working great for a _long_ time, but recently I noticed if I play a specific album, it resets and starts playing the same set of albums every time I clear the queue - ie it doesn't seem "random" at all. Anyone have any ideas on how to make the album selection more random? Thanks!

**Reply by jcorporation** [reply]
Can you provide a step by step guide to replicate the issue? You are using the latest myMPD version?

**Reply by chiefy** [reply]
Actually I think it's just user error somehow. I didn't know about the jukebox queue until poking around just now. I cleared it out and it seems more randomized now. Still kinda weird that it felt like that queue just kept resetting playing the same 10-12 albums.

**Comment by xavix** [comment]
In Jukebox mode, you can apply an exclusion filter and then save all the jukebox setting with a defined name. In example, you can set an exclusion filter like that: (Genre contains "Metal") OR (Genre contains "Crossover")... and save it as in the following screenshot:
<img width="898" height="160" alt="Captura de pantalla de 2026-03-14 09-32-16" src="https://github.com/user-attachments/assets/d6f20375-8264-40ab-b75d-c96ae4596bc0" />
But for any reason, this doesn't works for me. Random mode continues selecting songs that should be excluded by the filter. The include expression is empty. I'm using 24.0.3.
Thanks in advance!

**Reply by jcorporation** [reply]
OR is not supported: https://mpd.readthedocs.io/en/latest/protocol.html#filters

**Reply by xavix** [reply]
Thanks for the reply. It is possible to use regular expressions instead.

**Reply by jcorporation** [reply]
Yes, this should be possible, a include filter is usually the better attempt:
`((Genre != "Metal") AND (Genre != "Crossover"))`
